packages/app/src/Feed/BadgesFeed.ts

```diff
@@ -0,0 +1,99 @@
+import { useMemo } from "react";
+import { TaggedRawEvent, EventKind, HexKey, Lists, Subscriptions } from "@snort/nostr";
+import useSubscription from "Feed/Subscription";
+import { unwrap, findTag, chunks } from "Util";
+
+type BadgeAwards = {
+  pubkeys: string[];
+  ds: string[];
+};
+
+export default function useProfileBadges(pubkey?: HexKey) {
+  const sub = useMemo(() => {
+    if (!pubkey) return null;
+    const s = new Subscriptions();
+    s.Id = `profile_badges:${pubkey.slice(0, 12)}`;
+    s.Kinds = new Set([EventKind.ProfileBadges]);
+    s.DTags = new Set([Lists.Badges]);
+    s.Authors = new Set([pubkey]);
+    return s;
+  }, [pubkey]);
+  const profileBadges = useSubscription(sub, { leaveOpen: true, cache: false });
+
+  const profile = useMemo(() => {
+    const sorted = [...profileBadges.store.notes];
+    sorted.sort((a, b) => b.created_at - a.created_at);
+    const last = sorted[0];
+    if (last) {
+      return chunks(
+        last.tags.filter(t => t[0] === "a" || t[0] === "e"),
+        2
+      ).reduce((acc, [a, e]) => {
+        return {
+          ...acc,
+          [e[1]]: a[1],
+        };
+      }, {});
+    }
+    return {};
+  }, [pubkey, profileBadges.store.notes]);
+
+  const awardsSub = useMemo(() => {
+    if (!pubkey) return null;
```

Comment:
Add check to make sure `profile` has keys

Reply:
`Connection.ts:195 [wss://relay.nostr.band/] NOTICE: rejected: empty filter array`